CDCgov/prime-reportstream
# Experience/follow up for user state management refactors

This refactors the shape of values being sent down by the SessionContext to simplify the interface for interacting with session state.

Also deletes some unused utils, and moves / renames some files for clarity

This is a follow up to https://github.com/CDCgov/prime-reportstream/pull/6397, and does not include any functional changes.

Note that changes in the last commit are present for convenience of testing, and encapsulate the functional fixes from https://github.com/CDCgov/prime-reportstream/pull/6414 and https://github.com/CDCgov/prime-reportstream/pull/6416/

## Test Steps:

This change does not present any functional changes to the application. Please run a thorough smoke test of all parts of the application, with a particular focus on user auth, to ensure that this change is working properly.

## Changes
- *Include a comprehensive list of changes in this PR*
- *(For web UI changes) Include screenshots/video of changes*

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
